src/Transforms/BottomUp.dfy

```diff
@@ -30,10 +30,10 @@ module Bootstrap.Transforms.BottomUp {
     // - `e'` satisfies the pre of `f`
   {
     forall e, e' ::
-       && Exprs.ConstructorsMatch(e, e')
-       && f.requires(e)
-       && Deep.AllChildren_Expr(e', post)
-       ==> f.requires(e')
+    && Exprs.ConstructorsMatch(e, e')
+    && f.requires(e)
+    && Deep.AllChildren_Expr(e', post)
+    ==> f.requires(e')
```

Comment:
You don't indent the body of `forall`?

Reply:
forall statements, yes, but not forall expressions.
One reason for that is that forall takes everything afterwards.
But I add this to my final survey